spinalcordtoolbox/scripts/sct_maths.py

```diff
@@ -270,223 +315,465 @@ def __call__(self, _, namespace, values, option_string=None):
     similarity.add_argument(
         '-mi',
         metavar=Metavar.file,
+        action=StoreTodo,
         help='Compute the mutual information (MI) between both input files (-i and -mi) as in: '
              'https://scikit-learn.org/stable/modules/generated/sklearn.metrics.mutual_info_score.html',
         required=False)
     similarity.add_argument(
         '-minorm',
         metavar=Metavar.file,
+        action=StoreTodo,
         help='Compute the normalized mutual information (MI) between both input files (-i and -mi) as in: '
              'https://scikit-learn.org/stable/modules/generated/sklearn.metrics.normalized_mutual_info_score.html',
         required=False)
     similarity.add_argument(
         '-corr',
         metavar=Metavar.file,
-        help='Compute the cross correlation (CC) between both input files (-i and -cc).',
+        action=StoreTodo,
+        help='Compute the cross correlation (CC) between both input files (-i and -corr).',
         required=False)
 
     misc = parser.add_argument_group("MISC")
     misc.add_argument(
         '-symmetrize',
         type=int,
+        choices=(0, 1, 2),
+        action=StoreTodo,
         help='Symmetrize data along the specified dimension.',
-        required=False,
-        choices=(0, 1, 2))
+        required=False)
     misc.add_argument(
         '-type',
-        required=False,
+        choices=('uint8', 'int16', 'int32', 'float32', 'complex64', 'float64',
+                 'int8', 'uint16', 'uint32', 'int64', 'uint64'),
+        default=None,  # needed because argument_default=argparse.SUPPRESS
         help='Output type.',
-        choices=('uint8', 'int16', 'int32', 'float32', 'complex64', 'float64', 'int8', 'uint16', 'uint32', 'int64',
-                 'uint64'))
+        required=False)
     optional.add_argument(
         '-v',
         metavar=Metavar.int,
         type=int,
         choices=[0, 1, 2],
         default=1,
         # Values [0, 1, 2] map to logging levels [WARNING, INFO, DEBUG], but are also used as "if verbose == #" in API
-        help="Verbosity. 0: Display only errors/warnings, 1: Errors/warnings + info messages, 2: Debug mode")
+        help="Verbosity. 0: Display only errors/warnings, 1: Errors/warnings + info messages, 2: Debug mode",
+        required=False)
 
     return parser
 
 
-def apply_array_operation(data, dim, arg_name, arg_value, parser):
-    dim_list = ['x', 'y', 'z', 't']
+class SctMathsTypeError(Exception):
+    """Inappropriate argument type for an `sct_maths` operation."""
```

Comment:
I made custom exception classes so that we could catch them specifically, but maybe that's overkill.